File: 백준 10610번 30.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

string s;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cin>>s;
	int sum(0);
	for(char c: s)
	sum+=c-'0';
	sort(s.begin(),s.end());
	if(s[0]=='0'&&sum%3==0)
	{
		reverse(s.begin(),s.end());
		cout<<s;
	}
	else
	cout<<"-1";
} 

File: Boj-Kotlin&Java/백준 10610번 30.kt
package 백준

fun main() {
    val br=System.`in`.bufferedReader()
    val s=br.readLine()
    var list= arrayListOf<Int>()
    var sum=0
    for(i in s.indices){
        sum+=s[i]-'0'
        list.add(s[i]-'0')
    }
    list.sort()
    if(list[0]==0 && sum%3==0){
        list.reverse()
        for(i in list)
        print(i)
    }else
        println(-1)
}